active/0002-packager.md

```diff
@@ -0,0 +1,257 @@
+- __engines__: Once we have the concept of engines, developers will more than likely want to break apart into functional sets. This will allow you to build a bundle for each engine. These are looked at as entries to the pre-packager.
+- __http2__: As more servers move towards using [HTTP2](https://http2.github.io/) you actually want granular assets instead of concated blobs. This is due to the fact that you have [multiplexing](https://www.mnot.net/blog/2014/01/30/http2_expectations) in HTTP2. This allows the browser to cache bust only on the things that have changed. This may change to "none".
+- __custom__: If none of the out of the box steps work for you can tell the packager that you're are going to write your own concatenation logic.
+
+### Custom Concatenation
+As part of the options you can set the `composeOutput` property to a function that will receive the tree and the resolved dependency graph. It's then up to you to preform the concatenation and return the the tree.
+
+### Post Build Addon Hooks
+Once the tree has been concated, we then send that tree through the addon post-build hooks. This completes the the addon hook lifecycle.
+
+# Migration Plan
+
+## Ember CLI Developers
+
+We need to move all of the client dependencies that Ember CLI relies on to start publishing to npm as addons and are resolve their dependencies within node_modules as oppossed to bower_components. We will also need to rename some of the modules as they are at unreachable code paths e.g. `ember/resolver` does not actually resolve to the ember namespace.
+
+## Apps
+
+To ease the transition away from Bower we must provide an Ember CLI addon that adds a dependency resolver for bower components.  This would use the bower.json to whitelist modules and simply pull them into the tree from bower_components. We should warn whenever a dependency is resolved this way.
+
+Migrating to the Packager is as simple as changing the following in your Brocfile.
+
+```
+var Packager = require('ember-cli-packager');
+var packager = new Packager({
+  // Any exisiting EmberApp configuration plus the follow new options
+  entries: ['my-app'], // Optional if you're only building your app
+  strategies: ['default'], // Optional unless you want to change the concating
+  composeOutput: function(tree, graph) {} // Optional
+});
+```
+
+Since we are traversing a graph `app.import` for javascript dependencies should be removed.
+
+## Addons
+
+During this process we have slightly tweaked the structure of the app and addon folders in an addon.
+
+### App Directory
+For the app `pods` is now a directory and will get merged into the consumers namespace under `app/pods`.
+
+### Addon Directory
+Instead of only having 1 namespace per addon we have changed the directory structure to allow for N namespaces from an addon. 
+
+The new directory structure looks like the following:
+
+```
+my-addon/
+  addon/
+    my-namespaced-thing/
```

Comment:
also, `my-namespaced-thing.js` would be the `bucket` `import foo from 'my-namespaced-thing'` comes from.